### PR TITLE
Add BoundingBox section

### PR DESCRIPTION
**Pull request by ndunningbmw**
﻿#### Reference to a related issue in the repository
Add a reference to a related issue in the repository.

#### Add a description
Add a description of the changes proposed in the pull request.

**Some questions to ask**:
What is this change?
What does it fix?
Is this a bug fix or a feature? Does it break any existing functionality or force me to update to a new version?
How has it been tested?

#### Take this checklist as orientation for yourself, if this PR is ready for the Change Control Board:
- [ ] My suggestion follows the [style and contributors guidelines](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/start_contributing.html).
- [ ] I have taken care about the [message documentation](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/commenting_messages.html) and the [fields and enums documentation](https://opensimulationinterface.github.io/osi-antora-generator/asamosi/latest/specification/contributing/commenting_fields_enums.html).
- [ ] I have done the [DCO signoff](https://opensimulationinterface.github.io/osi-documentation/open-simulation-interface/doc/howtocontribute.html#developer-certification-of-origin-dco).
- [ ] My changes generate no errors when passing CI tests. 
- [ ] I have successfully implemented and tested my fix/feature locally.
- [ ] Appropriate reviewer(s) are assigned.

If you can’t check all of them, please explain why.
If all boxes are checked or commented and you have achieved at least one positive review, you can assign the label ReadyForCCBReview!

Signed-off-by: Nicholas Dunning <Nicholas.Dunning@bmw.de>

### CODE REVIEW

File: osi_common.proto [hunk 1]
```diff
@@ -447,6 +566,23 @@ message BaseStationary
     // The polygon is defined counter-clockwise.
     //
     repeated Vector2d base_polygon = 4;
+
+    // Sub-divisions of the overall bounding box of the \c BaseStationary object.
+    //
+    // The bounding box sections can include separate parts on partially-opaque
+    // objects such are trees with a distinction between trunk and crown.
```
Comment (PhRosenberger) [inline osi_common.proto:575]: Did you mean "such as trees"?

File: osi_common.proto [hunk 2]
```diff
@@ -387,6 +387,125 @@ message LogicalLaneAssignment
...
+
+    // Opaque reference of an associated 3D model of the bounding box.
+    //
+    // \note It is implementation-specific how model_references are resolved to
+    // 3d models. This means the coordinate system, model origin, and model
+    // orientation are also implementation-specific.
+    //
+    optional string model_reference = 5;
+
+    // Definition of different types of object contained within the bounding box
```
Comment (PhRosenberger) [inline osi_common.proto:434]: Is there a reference for this list or is it a result of a working-group brainsorming?
If there are sources, I qould like to see them referenced here.

File: osi_common.proto [hunk 3]
```diff
@@ -387,6 +387,125 @@ message LogicalLaneAssignment
     optional double angle_to_lane = 4;
 }
 
+// \brief A bounding box containing a sub-section of a object.
```
Comment (PhRosenberger) [inline osi_common.proto:390]: Typo: "of an object"

File: osi_common.proto [hunk 4]
```diff
@@ -387,6 +387,125 @@ message LogicalLaneAssignment
     optional double angle_to_lane = 4;
 }
 
+// \brief A bounding box containing a sub-section of a object. 
+//
+// A bounding box representing a sub-section of it's parents overall dimension, 
+// either that of a \c MovingObject or \c StationaryObject .
+//
+// The parent frame of the \c BoundingBox is not relative to the parent object
+// it is associated to, but in the same parent frame as the parent object.
```
Comment (PhRosenberger) [inline osi_common.proto:396]: I don't understand this sentence, could it be reworded?

**Review by PhRosenberger — COMMENTED** [review]
Is there a working example or demo? That would be great to see!
Nevertheless, this PR sounds perfectly reasonable to me.

File: osi_common.proto [hunk 5]
```diff
@@ -569,6 +707,23 @@ message BaseMoving
...
     repeated Vector2d base_polygon = 7;
+
+    // Sub-divisions of the overall bounding box of the \c BaseMoving object.
+    //
+    // The bounding box sections can include side mirrors, cargo, etc. for 
+    // vehicles, as well as body-part sections for pedestrians.
+    //
+    // \note When one or more \c BoundingBox s are associated to a 
+    // \c BaseMoving , the expectation is that all sections are contained
+    // within the bounds of the \c #dimension .
```
Comment (thomassedlmayer) [inline osi_common.proto:718]: Saying that bounding box sections can include side mirrors here but then saying they might not be included within the bounding box dimensions in the respective enum description (see line 459) seems kind of ambiguous and confusing to me.
Shouldn't we be clear about if we want them to be included or not? Otherwise some people include side mirrors, others don't...

I think the main questions are:
- Would it break backwards compatibility if we include side mirrors in the bounding box?
- If yes, do we want to change this for 4.0?

Based on that I would rephrase the descriptions to explain why it is as it is and maybe also state the goal for the next major release (if there is any).

File: osi_common.proto (hunk 4 above)
Comment (thomassedlmayer) [reply osi_common.proto:396]: What's actually meant is that a sub bounding box is not relative to its parent bounding box but rather to the parent frame of its parent bounding box. It is in the same frame as its parent object. E.g. if the main bounding box is in global coordinates, the sub bounding box is as well.

But I definitely agree, that it is hard to understand. Could also be extended by an example.

File: osi_common.proto [hunk 6]
```diff
@@ -569,6 +707,23 @@ message BaseMoving
...
+    // 
+    // \note It is also expcted that when using \c #bounding_box_section ,
```
Comment (thomassedlmayer) [inline osi_common.proto:720]: Typo "expcted"

File: osi_common.proto [hunk 7]
```diff
@@ -447,6 +568,23 @@ message BaseStationary
...
+    // Sub-divisions of the overall bounding box of the \c BaseStationary object.
+    //
+    // The bounding box sections can include separate parts on partially-opaque
+    // objects such are trees with a distinction between trunk and crown.
+    //
+    // \note When one or more \c BoundingBox s are associated to a 
+    // \c BaseStationary , the expectation is that all sections are contained
+    // within the bounds of the \c #dimension .
+    // 
+    // \note It is also expcted that when using \c #bounding_box_section ,
```
Comment (thomassedlmayer) [inline osi_common.proto:581]: Typo "expcted"

**Comment by ClemensLinnhoff** [comment]
There should also be harmonization with the model structure defined in [OpenMaterial](https://github.com/LudwigFriedmann/OpenMATERIAL/tree/master/model_structure) @LudwigFriedmann.
